src/Calculator/SettingsPage.xaml.cpp

```diff
@@ -0,0 +1,208 @@
+            {
+                Windows::Storage::ApplicationData::Current->LocalSettings->Values->Insert(L"CurrentTheme", L"Dark");
+            }
+        }
+        else if (colorSetting == L"System")
+        {
+            SettingsSystemTheme->IsChecked = true;
+            if (currentTheme == nullptr)
+            {
+                Windows::Storage::ApplicationData::Current->LocalSettings->Values->Insert(L"CurrentTheme", L"System");
+            }
+        }
+    }
+    else
+    {
+        SettingsSystemTheme->IsChecked = true;
+        if (currentTheme == nullptr)
+        {
+            Windows::Storage::ApplicationData::Current->LocalSettings->Values->Insert(L"CurrentTheme", L"System");
+        }
+    }
+
+    if (restartValue != nullptr)
+    {
+        if (restartApp == L"True")
+        {
+            SettingsRestartApp->Visibility = ::Visibility::Visible;
+        }
+        else if (restartApp == L"False")
+        {
+            SettingsRestartApp->Visibility = ::Visibility::Collapsed;
+        }
+    }
+    else
+    {
+        SettingsRestartApp->Visibility = ::Visibility::Collapsed;
+    }
+}
+
+void SettingsPage::SetDefaultFocus()
+{
+    SettingsLightTheme->Focus(::FocusState::Programmatic);
+}
+
+void SettingsPage::InitializeContributeTextBlock()
+{
+    auto resourceLoader = AppResourceProvider::GetInstance();
+    std::wstring contributeHyperlinkText = resourceLoader->GetResourceString("SettingsContribute")->Data();
+
+    // The resource string has '%HL%' wrapped around 'GitHub'
+    // Break the string and assign pieces appropriately.
+    static const std::wstring delimiter{ L"%HL%" };
+    static const size_t delimiterLength{ delimiter.length() };
+
+    // Find the delimiters.
+    size_t firstSplitPosition = contributeHyperlinkText.find(delimiter, 0);
+    assert(firstSplitPosition != std::wstring::npos);
+    size_t secondSplitPosition = contributeHyperlinkText.find(delimiter, firstSplitPosition + 1);
+    assert(secondSplitPosition != std::wstring::npos);
+    size_t hyperlinkTextLength = secondSplitPosition - (firstSplitPosition + delimiterLength);
+
+    // Assign pieces.
+    auto contributeTextBeforeHyperlink = ref new String(contributeHyperlinkText.substr(0, firstSplitPosition).c_str());
+    auto contributeTextLink = ref new String(contributeHyperlinkText.substr(firstSplitPosition + delimiterLength, hyperlinkTextLength).c_str());
+    auto contributeTextAfterHyperlink = ref new String(contributeHyperlinkText.substr(secondSplitPosition + delimiterLength).c_str());
+
+    ContributeRunBeforeLink->Text = contributeTextBeforeHyperlink;
+    ContributeRunLink->Text = contributeTextLink;
+    ContributeRunAfterLink->Text = contributeTextAfterHyperlink;
+
+}
+
+void SettingsPage::BackButtonClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    auto rootFrame = dynamic_cast<::Frame ^>(Window::Current->Content);
+
+    if (SettingsRestartApp->Visibility == ::Visibility::Visible)
```

Comment:
Might be better to do this at the time the user switches their selection instead to avoid writing this to settings every time they use the back button.